merkletree_test.go

```diff
@@ -0,0 +1,312 @@
+package merkletree
+
+import (
+	"bytes"
+	"crypto/rand"
+	"reflect"
+	"testing"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+
+	"golang.org/x/crypto/ed25519"
+	"golang.org/x/crypto/sha3"
+)
+
+var treeNonce = []byte("TREE NONCE")
+var salt = []byte("salt")
+
+var (
+	pk, sk []byte
+)
+
+func init() {
+	pk, sk, _ = ed25519.GenerateKey(rand.Reader)
```

Comment:
I think `init` is called by default by golang test suite.  The variable it's initializing `pk` and `sk` are used below at least.

The reason I'm asking this to be moved to the crypto pkg is because we are otherwise abstracting away the choices of crypto primitives.